Flexa-Agents/Large-deposits/src/agent.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  getEthersProvider,
+  ethers,
+} from "forta-agent";
+import { BigNumber, utils, providers } from "ethers";
+import abi from "./abi";
+
+const AMOUNT_THRESHOLD = utils.parseEther("1").mul(1e6); // 1 million
+const AMP_TOKEN: string = "0xfF20817765cB7f73d4bde2e66e067E58D11095C2";
+const FLEXA_CONTRACT: string = "0x706D7F8B3445D8Dfc790C524E3990ef014e7C578";
+
+export const createFinding = (
+  amount: BigNumber,
+  partition: string,
+  operator: string,
+  from: string,
+  destinationPartition: string,
+  to: string,
+  operatorData: string,
+): Finding => {
+  return Finding.fromObject({
+    name: "Large Deposit",
+    description: "Large Deposit into staking pool",
+    alertId: "FLEXA-2",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    metadata: {
+      value: amount.toString(),
+      fromPartition: partition.toLowerCase(),
+      operator: operator.toLowerCase(),
+      from: from.toLowerCase(),
+      destinationPartition: destinationPartition.toLowerCase(),
+      to: to.toLowerCase(),
+      operatorData: operatorData.toLowerCase(),
+    },
+  });
+};
+
+export function provideHandleTransaction(
+  amountThreshold: BigNumber,
+  ampToken: string,
+  flexaManager: string,
+  provider: providers.Provider,
+) {
+  const flexaStakingContract = new ethers.Contract(
+    flexaManager,
+    abi.COLLATERAL_MANAGER,
+    provider,
+  );
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+
+    // filter the transaction logs for transferByPartition events
+    const transferByPartitionEvents = txEvent.filterLog(
+      abi.AMP_TOKEN,
+      ampToken
+    );
+
+    // fire alerts for transfers of large stake
+    await Promise.all(transferByPartitionEvents.map(async (event) => {
+      const data = event.args.data;
+      const value = event.args.value;
+
+      //derives destinationAddress from data argument
+      const [,decodedPartition] = utils.defaultAbiCoder.decode(
+        ["bytes32", "bytes32"],
+        data,
+      );
+
+      const isValidPartition = await flexaStakingContract.partitions(
+        decodedPartition,
+        { blockTag: txEvent.blockNumber },
+      );
+
+
+      if (isValidPartition) {
+        if (value.gte(amountThreshold)) {
```

Comment:
We are using 1m USD as a threshold, consider fetching the AMP token price in order to convert `value` to USD before comparing it to the threshold.

Reply:
Good catch, I missed this one